--- drivers/hid/usbhid/hid-core.c.orig	2025-12-15 14:15:11.172818337 +0000
+++ drivers/hid/usbhid/hid-core.c	2025-12-15 14:17:48.325935240 +0000
@@ -985,6 +985,7 @@
 	struct usb_device *dev = interface_to_usbdev (intf);
 	struct hid_descriptor *hdesc;
 	struct hid_class_descriptor *hcdesc;
+        __u8 fixed_opt_descriptors_size;
 	u32 quirks = 0;
 	unsigned int rsize = 0;
 	char *rdesc;
@@ -1015,7 +1016,19 @@
 			      (hdesc->bNumDescriptors - 1) * sizeof(*hcdesc)) {
 		dbg_hid("hid descriptor invalid, bLen=%hhu bNum=%hhu\n",
 			hdesc->bLength, hdesc->bNumDescriptors);
-		return -EINVAL;
+                /*
+                 * Some devices may expose a wrong number of descriptors compared
+                 * to the provided length.
+                 * However, we ignore the optional hid class descriptors entirely
+                 * so we can safely recompute the proper field.
+                 */
+                if (hdesc->bLength >= sizeof(*hdesc)) {
+                    fixed_opt_descriptors_size = hdesc->bLength - sizeof(*hdesc);
+                    hid_warn(intf, "fixing wrong optional hid class descriptors count\n");
+                    hdesc->bNumDescriptors = fixed_opt_descriptors_size / sizeof(*hcdesc) + 1;
+                } else {
+                    return -EINVAL;
+                }
 	}
 
 	hid->version = le16_to_cpu(hdesc->bcdHID);
